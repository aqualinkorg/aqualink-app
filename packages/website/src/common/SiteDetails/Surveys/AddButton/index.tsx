import React from "react";
import { Link } from "react-router-dom";
import {
  Button,
  Typography,
  withStyles,
  WithStyles,
  createStyles,
} from "@material-ui/core";
import AddCircleOutlineIcon from "@material-ui/icons/AddCircleOutline";
import { grey } from "@material-ui/core/colors";

const AddButton = ({ siteId, classes }: AddButtonProps) => {
  return (
    <Button
      className={classes.addSurveyButton}
      startIcon={<AddCircleOutlineIcon />}
      component={Link}
      to={`/sites/${siteId}/new_survey`}
      classes={{ iconSizeMedium: classes.addSurveyButtonIcon }}
    >
      <Typography color="inherit" variant="h6">
        ADD NEW SURVEY
      </Typography>
    </Button>
  );
};

const styles = () =>
  createStyles({
    addSurveyButton: {
      color: grey[500],
      "&:hover": {
        color: grey[500],
      },
    },
    addSurveyButtonIcon: {
      "& > *:first-child": {
        fontSize: 32,
      },
    },
  });

interface AddButtonIncomingProps {
  siteId: number;
}

type AddButtonProps = AddButtonIncomingProps & WithStyles<typeof styles>;

export default withStyles(styles)(AddButton);
