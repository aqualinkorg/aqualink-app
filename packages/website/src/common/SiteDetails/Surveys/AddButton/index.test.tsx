import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import AddButton from '.';

test('renders as expected', () => {
  const { container } = render(
    <Router>
      <AddButton siteId={1} />
    </Router>,
  );
  expect(container).toMatchSnapshot();
});
