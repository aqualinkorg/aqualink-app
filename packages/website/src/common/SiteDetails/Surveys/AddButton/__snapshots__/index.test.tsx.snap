// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as expected 1`] = `
<div>
  <mock-button
    classes="[object Object]"
    classname="AddButton-addSurveyButton-1"
    component="[object Object]"
    starticon="[object Object]"
    to="/sites/1/new_survey"
  >
    <mock-typography
      color="inherit"
      variant="h6"
    >
      ADD NEW SURVEY
    </mock-typography>
  </mock-button>
</div>
`;
